ci/fireci/fireciplugins/macrobenchmark.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import glob
+import json
+import logging
+import os
+import random
+import sys
+
+import pystache
+import shutil
+import yaml
+
+from fireci import ci_command
+from fireci.dir_utils import chdir
+
+_logger = logging.getLogger('fireci.macrobenchmark')
+
+
+@ci_command()
+def macrobenchmark():
+  """Measures app startup times for Firebase SDKs."""
+  asyncio.run(_launch_macrobenchmark_test())
+
+
+async def _launch_macrobenchmark_test():
+  _logger.info('Starting macrobenchmark test...')
+
+  artifact_versions, config, _ = await asyncio.gather(
+    _parse_artifact_versions(),
+    _parse_config_yaml(),
+    _create_gradle_wrapper()
+  )
+
+  with chdir('macrobenchmark'):
+    runners = [MacrobenchmarkTest(k, v, artifact_versions) for k, v in config.items()]
+    await asyncio.gather(*[x.run() for x in runners])
+
+  _logger.info('Macrobenchmark test finished.')
+
+
+async def _parse_artifact_versions():
+  proc = await asyncio.subprocess.create_subprocess_exec('./gradlew', 'assembleAllForSmokeTests')
+  await proc.wait()
```

Comment:
wouldn't line 56 await be enough? not sure

Reply:
Sounds to me it is required. So if I don't `await` on it, it gives me this warning:
```
/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/unix_events.py:878: RuntimeWarning: A loop is being detached from a child watcher with pending handlers
  RuntimeWarning)
```
My understanding is that it means the underneath [`Future` object](https://docs.python.org/3/library/asyncio-future.html#future-object) may not be settled before my code exit. Also in the [python doc](https://docs.python.org/3/library/asyncio-subprocess.html) for asyncio subprocess, the first [example](https://docs.python.org/3/library/asyncio-subprocess.html#subprocesses) and the last [example](https://docs.python.org/3/library/asyncio-subprocess.html#examples) also `await` on `proc.communicate()` and `proc.wait()` respectively.

This also reminds me that I forgot to await at other places in the code. The problem was masked previously. I fixed in the new commit.